pandas/core/arrays/interval.py

```diff
@@ -222,13 +228,14 @@ def _simple_new(
 
         if dtype is not None:
             # GH 19262: dtype must be an IntervalDtype to override inferred
-            dtype = pandas_dtype(dtype)
```

Comment:
why did you remove this convesion?

Reply:
Based on the logic and usage of `dtype` it seemed unnecessary. I wanted to check if all the tests will pass, if not then I will again revert it.